### PR TITLE
WEB-10426

### PR DESCRIPTION
**Pull request by matiaslg13**
- Including new `PUT` endpoint and updated `GET` endpoint

### CODE REVIEW

File: types/Identity.d.ts [hunk 1]
```diff
@@ -26,6 +26,9 @@ export default class IdentityService extends Service {
         emailAddress: string;
         redirectUri?: string;
     }): Promise<any>;
+    updateUser({ emailAddress }: {
+        emailAddress: string;
+    }): Promise<any>;
```
Comment (AnnieCSerato) [inline types/Identity.d.ts:31]: Is the return type suppose to be `Promise<User>`?

**Review by Krithik-Serato — COMMENTED** [review]
Does this PR also include changes for the new `GET /users` endpoint?

File: types/Identity.d.ts (hunk 1 above)
Comment (matiaslg13) [reply types/Identity.d.ts:31]: I think so, I was confused by the previous one `changeEmailAddress` wich type is  `Promise<any>`

**Comment by matiaslg13** [comment]
> Will this PR also include changes for the new `GET /users` endpoint?

That endpoint already exists, what would we need to update here in that case. I mean, apart from what I've updated on the export type `User` to include `email_address_history` ?

**Comment by Krithik-Serato** [comment]
> > Will this PR also include changes for the new `GET /users` endpoint?
> 
> That endpoint already exists, what would we need to update here in that case. I mean, apart from what I've updated on the export type `User` to include `email_address_history` ?

Currently it only returns a single user, so wouldn't we need to return an array object of Users? I was thinking something like: 
`getUsers({ emailAddress }: {
        emailAddress: string;
    }): Promise<Identity.UserList>;`

File: types/Identity.d.ts (hunk 1 above)
Comment (AnnieCSerato) [reply types/Identity.d.ts]: tbh that previous one should probably be just `Promise` but 🤷‍♀️

Comment (matiaslg13) [reply types/Identity.d.ts]: Oh right 🤔

File: test/integration/IdentityTest.js [hunk 2]
```diff
@@ -105,5 +105,23 @@ describe('Identity Tests', function () {
           }
       )
     })
+    it(`confirms URI in 'getUsers()' method without user ID, by returning a non-404 response`, function () {
+      swsClient.userId = 0
```
Comment (AnnieCSerato) [inline test/integration/IdentityTest.js:109]: Do we need two tests for this endpoint if there isn't a difference in behaviour if the userId is set or not?

File: types/Identity.d.ts [hunk 3]
```diff
@@ -67,5 +70,12 @@ export namespace Identity {
     export type OkMessage = {
         message: string;
     };
+    export type UserUpdateParams = {
```
Comment (AnnieCSerato) [inline types/Identity.d.ts:73]: is this used?

Comment (Krithik-Serato) [reply types/Identity.d.ts:73]: Oh yep this one isn't needed. I can move this to `modules/identity.ts` in the console app repo

File: test/integration/IdentityTest.js (hunk 2 above)
Comment (matiaslg13) [reply test/integration/IdentityTest.js]: Good point, just updated it!

File: types/Identity.d.ts (hunk 3 above)
Comment (matiaslg13) [reply types/Identity.d.ts]: After confirming with @Krithik-Serato , we can remove it here.

**Pull request closed without merging**